### PR TITLE
Ran clang-tidy over the source tree

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

File: tests/tools/plugins/ssl_hook_test.cc [hunk 1]
```diff
@@ -332,7 +332,7 @@ setup_callbacks(TSHttpTxn txn, int preaccept_count, int client_hello_count, int
   }
   for (i = 0; i < sni_count; i++) {
     cb = TSContCreate(&CB_SNI, nullptr);
-    TSContDataSet(cb, (void *)(intptr_t)i);
+    TSContDataSet(cb, (void *)static_cast<intptr_t>(i));
```
Comment (bneradt) [inline tests/tools/plugins/ssl_hook_test.cc:335]: I'm surprised this doesn't apply a C++ cast for the (void *) c-style cast. Do you know why it doesn't?

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by bryancall** [comment]
[approve ci clang-analyzer]

File: tests/tools/plugins/ssl_hook_test.cc (hunk 1 above)
Comment (bryancall) [reply tests/tools/plugins/ssl_hook_test.cc:335]: Don't know

**Pull request merged**